tests/pyfunc/test_spark.py

```diff
@@ -121,6 +122,39 @@ def test_spark_udf(spark, model_path):
                 assert expected == actual
 
 
+def test_spark_udf_autofills_no_arguments(spark, model_path):
+    class TestModel(PythonModel):
+        def predict(self, context, model_input):
+            return [model_input.columns] * len(model_input)
+
+    signature = ModelSignature(
+        inputs=Schema([ColSpec("long", "a"), ColSpec("long", "b"), ColSpec("long", "c")]),
+        outputs=Schema([ColSpec("integer")]),
+    )
+
+    with mlflow.start_run() as run:
+        mlflow.pyfunc.log_model("model", python_model=TestModel(), signature=signature)
+        udf = mlflow.pyfunc.spark_udf(
+            spark, "runs:/{}/model".format(run.info.run_id), result_type=ArrayType(StringType())
+        )
+        good_data = spark.createDataFrame(
+            pd.DataFrame(
+                columns=["a", "b", "c", "d"], data={"a": [1], "b": [2], "c": [3], "d": [4]}
+            )
+        )
+        res = good_data.withColumn("res", udf()).select("res").toPandas()
+        assert res["res"][0] == ["a", "b", "c"]
+
+        # this dataframe won't work because it's missing column a
+        bad_data = spark.createDataFrame(
+            pd.DataFrame(
+                columns=["x", "b", "c", "d"], data={"x": [1], "b": [2], "c": [3], "d": [4]}
+            )
+        )
+        with pytest.raises(AnalysisException, match=r"cannot resolve '`a`' given input columns"):
+            bad_data.withColumn("res", udf())
+
```

Comment:
Can we add a test case for what happens if the model has no signature and user calls it with empty argument list?